deals/module/deals.go

```diff
@@ -146,9 +146,11 @@ func (m *Module) Store(ctx context.Context, waddr string, dataCid cid.Cid, piece
 			Data: &storagemarket.DataRef{
 				TransferType: storagemarket.TTGraphsync,
 				Root:         dataCid,
+				PieceCid:     &pieceCid,
+				PieceSize:    pieceSize.Unpadded(),
```

Comment:
We delegate to the client to provide `pieceSize` and `pieceCid`. This allows Lotus to avoid calculating these values, and just trusting they're correct. This is important for clients such as FFS, which create multiple deals with the same data.